File: src/Product/Lothal.Product.Domain/Entities/Product.cs
namespace Lothal.Product.Domain.Entities;

public class Product
{
    public string Id { get; set; } = string.Empty; // Mapped from Barcode usually, required by ES convention
    public string Barcode { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Class { get; set; } = string.Empty;
    public string Color { get; set; } = string.Empty;
    public string Size { get; set; } = string.Empty;
}

File: src/UI/lothal-admin-ui/src/types/product.ts
export interface Product {
  barcode: string;
  name: string;
  class: string;
  color: string;
  size: string;
  price: number;
  id?: string;
}

export interface Stock {
  barcode: string;
  warehouseQuantity: number;
  availableQuantity?: number;
  source: string;
  lastUpdatedAt: string;
}

export interface ProductWithStock extends Product {
  stock?: Stock;
}
